src/conn.rs

```diff
@@ -342,11 +353,14 @@ mod tests {
         let t2 = "[{:db.schema/attribute \"three\", :db/ident :a/keyword1}]";
 
         let in_progress = conn.begin_transaction(&mut sqlite).expect("begun successfully");
-        let (in_progress, report) = in_progress.transact(t).expect("transacted successfully");
-        let one = report.tempids.get("one").cloned();
-        let (in_progress, report) = in_progress.transact(t2).expect("transacted again");
-        let mut three = report.tempids.get("three").cloned();
-        in_progress.commit().expect("Commit succeeded.");
+        let in_progress = in_progress.transact(t).expect("transacted successfully");
+        let one = in_progress.last_report().unwrap().tempids.get("one").cloned();
```

Comment:
There are helpers to assert the shape of `tempids`, IIRC.  And it would be good to ensure you're really getting new entity IDs (i.e., that the metadata updating is correct).